provisioning/service/src/Authentication/CredentialsWithTokenRefresh.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Provisioning.Service
+{
+    /// <summary>
+    /// Authentication class implementing SAS token refresh
+    /// </summary>
+    public abstract class CredentialsWithTokenRefresh: ProvisioningServiceClientCredentials, IDisposable
```

Comment:
This code is copy / paste from AuthenticationWithToken refresh. Please unify or leave a TODO if you can't get to it within this PR. 
We already have a lot of code duplication - let's not add more.